components/wallet-index-check.tsx

```diff
@@ -0,0 +1,122 @@
+"use client"
+
+import { useEffect, useId } from "react"
+
+import {
+  ConfigRequiredError,
+  useConfig,
+  useWasmInitialized,
+} from "@renegade-fi/react"
+import {
+  createWallet,
+  getBackOfQueueWallet,
+  lookupWallet,
+} from "@renegade-fi/react/actions"
+import { useMutation } from "@tanstack/react-query"
+import { Loader2 } from "lucide-react"
+import { toast } from "sonner"
+
+import { useServerStore } from "@/providers/state-provider/server-store-provider"
+
+// Toast messages
+const MSG_INITIAL = "Hang tight…setting up your wallet"
+const MSG_CREATING = "Creating your wallet…"
+const MSG_RECOVER = "Recovering your wallet…"
+const MSG_SUCCESS = "All set! Your wallet is ready"
+const MSG_ERROR = "Oops, something went wrong"
+
+// Polling constants
+const POLL_DELAY_MS = 2000
+const POLL_TIMEOUT_MS = 60000
+
+export function WalletIndexCheck() {
+  const seed = useServerStore((s) => s.wallet.seed)
+  const chainId = useServerStore((s) => s.wallet.chainId)
+  const walletId = useServerStore((s) => s.wallet.id)
+  const wasmInitialized = useWasmInitialized()
+  const config = useConfig()
+  const toastId = useId()
+
+  async function delay(ms: number) {
+    return new Promise<void>((resolve) => setTimeout(resolve, ms))
+  }
+
+  async function indexWallet() {
+    if (!config) throw new ConfigRequiredError("indexWallet")
+
+    try {
+      // Step 1: Check if wallet already exists
+      try {
+        const existing = await getBackOfQueueWallet(config)
```

Comment:
nit: we could just `getWallet`, it's technically a little faster than `getBackOfQueueWallet` and all we need to know is if the wallet is in the relayer, not its state